numpy/core/tests/test_function_base.py

```diff
@@ -113,6 +113,26 @@ def test_basic(self):
         assert_array_equal(y, [-100, -10, -1])
         assert_array_equal(y.imag, 0)
 
+    def test_boundaries_match_start_and_stop_exactly(self):
```

Comment:
Could do
```suggestion
    @pytest.mark.parametrize('start_sign', [1, -1])
    @pytest.mark.parametrize('stop_sign', [1, -1])
    @pytest.mark.parametrize('num', [1, 3])
    def test_boundaries_match_start_and_stop_exactly(self, start_sign, stop_sign, num):
```
and adjust the test below accordingly

Reply:
Note sure that improve readability...?

(Let's hope quay.io comes back online anytime soon.)

Reply:
Ah, this actually catches an additional bug - `np.geomspace(-3, 3, num=3)` should give `[-3, nan, 3]`, not `[nan, nan, 3]`

Reply:
True, good catch - shall I amend the unit tests accordingly?

Reply:
Yes please!

Reply:
Done, please review.

Reply:
That additional test leads to a `RuntimeWarning` because it computes `log10(0.0)` which in turn seems to make the test fail. Why does a Warning lead to a test failure please? How can I avoid this?

Reply:
you can use something like `with np.errstate(divide="raise"):` and check that the floating point error is raised. if you want to ignore you can do something like `with np.errstate(divide="ignore"):`

Reply:
done, thanks